dynamic_programming/cut_rod.cpp

```diff
@@ -68,17 +68,71 @@ int maxProfitByCuttingRod(const std::array<int, T> &price, const int n) {
  * @brief Function to test above algorithm
  * @returns void
  */
+
+
+
+template <size_t T>
+bool WhatIfAllPricesAreSame(const std::array<int, T> &price, const int n){
```

Comment:
This function should be inside the namespaces above.
It is also recommended to use `uint64_t` in the place of `int`.
```suggestion
bool whatIfAllPricesAreSame(const std::array<int, T> &price, const uint64_t &n){
```
This function is also missing proper documentation (see #962 as a reference).

Reply:
i have done changes as requested, please take a look.